awx_collection/plugins/modules/group.py

```diff
@@ -129,6 +129,7 @@ def main():
     state = module.params.pop('state')
     preserve_existing_hosts = module.params.get('preserve_existing_hosts')
     preserve_existing_children = module.params.get('preserve_existing_groups')
+    preserve_existing_children = preserve_existing_children or module.params.get('preserve_existing_children')
```

Comment:
If a task in a playbook uses a module argument's alias, the value given to that alias will [get assigned](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/common/parameters.py#L235) to the original argument that declared that alias.

In other words, if `preserve_existing_children` declares a `preserve_existing_groups` alias and someone passes in `preserve_existing_groups: foo`, then `"foo"` will live in `module.params['preserve_existing_children']`.

So I think an alternative/better fix here would be to just use the _actual_ parameter name instead of its alias when we assign the value to `preserve_existing_children` on line 131.

Reply:
Thank you so much, @relrod! That makes sense to me! I have revised this per your suggestion. Please let me know if I need to make any additional changes!

Reply:
After that change, you should be able to remove line 132 entirely :)

Reply:
Oops, thank you!